70_Pushover.pm

```diff
@@ -235,80 +245,100 @@ sub Pushover_CGI() {
+                    }
+                    else {
+                        readingsBulkUpdate( $hash, $rAckAt, int( time() ) );
+                    }
                 }
                 else {
-                    readingsBulkUpdate( $hash, $rAckAt, int( time() ) );
-                }
 
-                my $redirect = "";
+                    my $redirect = "";
 
-                # run FHEM command if desired
-                if ( ReadingsVal( $name, $rAct, "pushover://" ) !~
-                    /^[\w-]+:\/\/.*$/ )
-                {
-                    $redirect = "pushover://";
-
-                    fhem ReadingsVal( $name, $rAct, "" );
-                    readingsBulkUpdate( $hash, $rAct,
-                        "executed: " . ReadingsVal( $name, $rAct, "" ) );
-                }
+                    # run FHEM command if desired
+                    if ( ReadingsVal( $name, $rAct, "pushover://" ) !~
+                        /^[\w-]+:\/\/.*$/ )
+                    {
+                        $redirect = "pushover://";
 
-                # redirect to presented URL
-                if ( ReadingsVal( $name, $rAct, "none" ) =~ /^[\w-]+:\/\/.*$/ )
-                {
-                    $redirect = ReadingsVal( $name, $rAct, "" );
-                }
+                        if (ReadingsVal( $name, $rAct, "" ) !~ /^executed: |^$/) {
+                            fhem ReadingsVal( $name, $rAct, "" );
+                            readingsBulkUpdate( $hash, $rAct,
+                                "executed: " . ReadingsVal( $name, $rAct, "" ) );
+                        }
+                    }
 
-                readingsEndUpdate( $hash, 1 );
+                    # redirect to presented URL
+                    if ( ReadingsVal( $name, $rAct, "none" ) =~ /^[\w-]+:\/\/.*$/ )
+                    {
+                        $redirect = ReadingsVal( $name, $rAct, "" );
+                    }
 
-                return (
-                    "text/html; charset=utf-8",
-                    "<html><head><meta http-equiv=\"refresh\" content=\"0;url="
-                      . $redirect
-                      . "\"></head><body><a href=\""
-                      . $redirect
-                      . "\">Click here to get redirected to your destination"
-                      . "</a></body></html>"
-                ) if ( $redirect ne "" );
+                    readingsEndUpdate( $hash, 1 );
+
+                    return (
+                        "text/html; charset=utf-8",
+                        "<html><head><meta http-equiv=\"refresh\" content=\"0;url="
+                          . $redirect
+                          . "\"></head><body><a href=\""
+                          . $redirect
+                          . "\">Click here to get redirected to your destination"
+                          . "</a></body></html>"
+                    ) if ( $redirect ne "" );
+                }
```

Comment:
Most of this is just indentation due to
2) A message is now only acknowledge after using the "Acknowledge" button within the Pushover notification. Clicking the link for an action will not automatically acknowledge the message.
The "else" part is what is being executed acknowledge flag